src/DataProtection/DataProtection/src/Repositories/EphemeralXmlRepository.cs

```diff
@@ -54,4 +54,59 @@ public void StoreElement(XElement element, string friendlyName)
             _storedElements.Add(cloned);
         }
     }
+
+    /// <inheritdoc/>
+    public bool RemoveElements(Action<IReadOnlyCollection<IDeletableElement>> chooseElements)
+    {
+        ArgumentNullThrowHelper.ThrowIfNull(chooseElements);
+
+        var deletableElements = new List<DeletableElement>();
+
+        foreach (var storedElement in _storedElements)
```

Comment:
This needs to be inside `lock (_storedElements)` or risk an `InvalidOperationException` for enumerating the collection during modification.

Reply:
Done, though I still think implemeting deletion on the ephemeral repository is silly.